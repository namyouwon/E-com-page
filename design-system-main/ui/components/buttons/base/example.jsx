// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { CannotBeSetWith } from '../../../shared/helpers';

const KX_SCOPE_BUTTON = 'button';
const KX_TYPE_RIPPLE = 'ripple';
const KX_TYPE_UNDERLINE = 'underline';

export const Button = props => {
  const {
    className,
    disabled,
    isReset,
    isBrand,
    isNeutral,
    isOutlineBrand,
    isInverse,
    isDestructive,
    isTextDestructive,
    isSuccess,
    isFullWidth,
    isStretch,
    isFirst,
    isMiddle,
    isLast,
    onClick,
    kxScope,
    kxType,
    use,
    innerRef,
    ...rest
  } = props;
  const pointerRef = { current: { offsetX: '50%', offsetY: '50%' } };
  const requestRef = { current: null };
  const previousTimeRef = { current: null };
  const buttonRef = innerRef || { current: null };
  const kxTypeUnderline = kxType && kxType === KX_TYPE_UNDERLINE;
  const kxTypeRipple = kxType && kxType === KX_TYPE_RIPPLE;

  let classNameList = classNames(
    'slds-button',
    {
      'slds-button_reset': isReset,
      'slds-button_brand': isBrand,
      'slds-button_neutral': isNeutral,
      'slds-button_outline-brand': isOutlineBrand,
      'slds-button_inverse': isInverse,
      'slds-button_destructive': isDestructive,
      'slds-button_text-destructive': isTextDestructive,
      'slds-button_success': isSuccess,
      'slds-button_full-width': isFullWidth,
      'slds-button_stretch': isStretch,
      'slds-button_first': isFirst,
      'slds-button_middle': isMiddle,
      'slds-button_last': isLast
    },
    className
  );

  const setCoordProps = element => {
    const { offsetX, offsetY } = pointerRef.current;
    element.style.setProperty(
      '--slds-kx-button-pointer-position-x',
      `${offsetX}px`
    );
    if (!kxTypeUnderline)
      element.style.setProperty(
        '--slds-kx-button-pointer-position-y',
        `${offsetY}px`
      );
  };

  const handleMove = ({ offsetX, offsetY }) => {
    pointerRef.current = { offsetX, offsetY };
  };

  const handleEnter = () => {
    requestRef.current = window.requestAnimationFrame(timestamp => {
      animate(timestamp, buttonRef.current);
    });
    buttonRef.current.addEventListener('mousemove', handleMove);
  };

  const handleLeave = () => {
    window.cancelAnimationFrame(requestRef.current);
    buttonRef.current.removeEventListener('mousemove', handleMove);
  };

  const animate = (timestamp, element) => {
    if (previousTimeRef.current !== undefined) setCoordProps(element);
    previousTimeRef.current = timestamp;
    requestRef.current = window.requestAnimationFrame(timestamp => {
      animate(timestamp, element);
    });
  };

  const setIsRippling = (element, doRipple) => {
    element.classList.toggle('slds-kx-is-animating-from-click', doRipple);
  };

  const handleClick = event => {
    const { target: buttonElement } = event;
    event.persist();
    if (onClick) onClick(event);
    if (buttonElement.tagName === 'A') event.preventDefault();
    if (!kxTypeRipple) return; // conditionally short circuit the handler since it is not kinetics type ripple
    const { offsetX, offsetY } = event.nativeEvent;
    pointerRef.current = { offsetX, offsetY };
    setCoordProps(buttonElement);
    setIsRippling(buttonElement, true);
  };

  const handleRippleEnd = event => {
    setIsRippling(event.target, false);
  };

  return (
    <React.Fragment>
      {use === 'a' ? (
        <a
          ref={buttonRef}
          className={classNameList}
          href="#"
          {...rest}
          onClick={handleClick}
          {...kxScope && { 'kx-scope': kxScope }}
          {...kxType && { 'kx-type': kxType }}
          {...kxScope &&
            kxScope.startsWith(KX_SCOPE_BUTTON) && {
              onMouseEnter: handleEnter,
              onMouseLeave: handleLeave
            }}
          {...kxType &&
            kxType === KX_TYPE_RIPPLE && { onAnimationEnd: handleRippleEnd }}
        >
          {props.children}
        </a>
      ) : (
        <button
          ref={buttonRef}
          className={classNameList}
          disabled={disabled}
          {...rest}
          onClick={handleClick}
          {...kxScope && { 'kx-scope': kxScope }}
          {...kxType && { 'kx-type': kxType }}
          {...kxScope &&
            kxScope.startsWith(KX_SCOPE_BUTTON) && {
              onMouseEnter: handleEnter,
              onMouseLeave: handleLeave
            }}
          {...kxType &&
            kxType === KX_TYPE_RIPPLE && { onAnimationEnd: handleRippleEnd }}
        >
          {props.children}
        </button>
      )}
    </React.Fragment>
  );
};

Button.displayName = 'Button';

Button.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  isReset: PropTypes.bool,
  isBrand: PropTypes.bool,
  isNeutral: PropTypes.bool,
  isOutlineBrand: PropTypes.bool,
  isInverse: PropTypes.bool,
  isDestructive: PropTypes.bool,
  isTextDestructive: PropTypes.bool,
  isSuccess: PropTypes.bool,
  isFullWidth: PropTypes.bool,
  isStretch: PropTypes.bool,
  isFirst: PropTypes.bool,
  isMiddle: PropTypes.bool,
  isLast: PropTypes.bool,
  kxScope: PropTypes.string,
  kxType: PropTypes.string,
  use: CannotBeSetWith('disabled', PropTypes.oneOf(['a'])),
  children: PropTypes.node
};

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default <Button>Button</Button>;

export let examples = [
  {
    id: 'base-kinetics',
    label: 'Base (Kinetics)',
    element: (
      <Button kxScope="button" kxType="underline">
        Kinetics Button
      </Button>
    )
  },
  {
    id: 'neutral',
    label: 'Neutral',
    element: <Button isNeutral>Neutral Button</Button>
  },
  {
    id: 'neutral-kinetics',
    label: 'Neutral (Kinetics)',
    element: (
      <Button isNeutral kxScope="button-neutral" kxType="ripple">
        Kinetics Neutral Button
      </Button>
    )
  },
  {
    id: 'brand',
    label: 'Brand',
    element: <Button isBrand>Brand Button</Button>
  },
  {
    id: 'brand-kinetics',
    label: 'Brand (Kinetics)',
    element: (
      <Button isBrand kxScope="button-brand" kxType="ripple">
        Kinetics Brand Button
      </Button>
    )
  },
  {
    id: 'outline-brand',
    label: 'Outline Brand',
    element: <Button isOutlineBrand>Outline Brand Button</Button>
  },
  {
    id: 'outline-brand-kinetics',
    label: 'Outline Brand (Kinetics)',
    element: (
      <Button isOutlineBrand kxScope="button-outline" kxType="ripple">
        Kinetics Outline Brand Button
      </Button>
    )
  },
  {
    id: 'inverse',
    label: 'Inverse',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button isInverse>Inverse Button</Button>
      </div>
    )
  },
  {
    id: 'inverse-kinetics',
    label: 'Inverse (Kinetics)',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button isInverse kxScope="button-neutral" kxType="ripple">
          Kinetics Inverse Button
        </Button>
      </div>
    )
  },
  {
    id: 'destructive',
    label: 'Destructive',
    element: <Button isDestructive>Destructive Button</Button>
  },
  {
    id: 'destructive-kinetics',
    label: 'Destructive (Kinetics)',
    element: (
      <Button isDestructive kxScope="button-filled" kxType="ripple">
        Kinetics Destructive Button
      </Button>
    )
  },
  {
    id: 'text-destructive',
    label: 'Text Destructive',
    element: <Button isTextDestructive>Text Destructive Button</Button>
  },
  {
    id: 'success',
    label: 'Success',
    element: <Button isSuccess>Success Button</Button>
  },
  {
    id: 'success-kinetics',
    label: 'Success (Kinetics)',
    element: (
      <Button isSuccess kxScope="button-filled" kxType="ripple">
        Kinetics Success Button
      </Button>
    )
  },
  {
    id: 'stretch',
    label: 'Stretch',
    element: (
      <Button isStretch isNeutral>
        Stretched Neutral Button
      </Button>
    )
  },
  {
    id: 'stretch-kinetics',
    label: 'Stretch (Kinetics)',
    element: (
      <Button isStretch isNeutral kxScope="button-neutral" kxType="ripple">
        Kinetics Stretched Neutral Button
      </Button>
    )
  },
  {
    id: 'full-width',
    label: 'Full Width',
    element: (
      <Button isFullWidth isNeutral>
        Full Width Neutral Button
      </Button>
    )
  },
  {
    id: 'base-link',
    label: 'Base as a Link',
    element: <Button use="a">Base Link</Button>
  },
  {
    id: 'neutral-link',
    label: 'Neutral as a Link',
    element: (
      <Button use="a" isNeutral>
        Neutral Link
      </Button>
    )
  },
  {
    id: 'brand-link',
    label: 'Brand as a Link',
    element: (
      <Button use="a" isBrand>
        Brand Link
      </Button>
    )
  },
  {
    id: 'outline-brand-link',
    label: 'Outline Brand as a Link',
    element: (
      <Button use="a" isOutlineBrand>
        Outline Brand Link
      </Button>
    )
  },
  {
    id: 'inverse-link',
    label: 'Inverse as a Link',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button use="a" isInverse>
          Inverse Link
        </Button>
      </div>
    )
  },
  {
    id: 'destructive-link',
    label: 'Destructive as a Link',
    element: (
      <Button use="a" isDestructive>
        Destructive Link
      </Button>
    )
  },
  {
    id: 'text-destructive-link',
    label: 'Text Destructive as a Link',
    element: (
      <Button use="a" isTextDestructive>
        Text Destructive Link
      </Button>
    )
  },
  {
    id: 'success-link',
    label: 'Success as a Link',
    element: (
      <Button use="a" isSuccess>
        Success Link
      </Button>
    )
  },
  {
    id: 'stretch-link',
    label: 'Stretch as a Link',
    element: (
      <Button use="a" isStretch isNeutral>
        Stretched Neutral Link
      </Button>
    )
  },
  /**
   * This is not best practice, use .slds-button_full-width or .slds-button_stretch
   * instead, this accounts for test coverage of legacy overrides by customers.
   */
  {
    id: 'base-centering',
    label: 'Base Centering',
    element: (
      <Button isNeutral style={{ width: '400px' }}>
        I should be centered
      </Button>
    )
  },
  {
    id: 'reset',
    label: 'Reset',
    element: <Button isReset>Reset Button</Button>
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Disabled',
    element: <Button disabled>Button</Button>
  },
  {
    id: 'base-kinetics-disabled',
    label: 'Base (Kinetics) - Disabled',
    element: (
      <Button kxScope="button" kxType="underline" disabled>
        Kinetics Button
      </Button>
    )
  },
  {
    id: 'neutral-disabled',
    label: 'Neutral - Disabled',
    element: (
      <Button isNeutral disabled>
        Neutral Button
      </Button>
    )
  },
  {
    id: 'neutral-kinetics-disabled',
    label: 'Neutral (Kinetics) - Disabled',
    element: (
      <Button isNeutral kxScope="button-neutral" kxType="ripple" disabled>
        Kinetics Neutral Button
      </Button>
    )
  },
  {
    id: 'brand-disabled',
    label: 'Brand - Disabled',
    element: (
      <Button isBrand disabled>
        Brand Button
      </Button>
    )
  },
  {
    id: 'brand-kinetics-disabled',
    label: 'Brand (Kinetics) - Disabled',
    element: (
      <Button isBrand kxScope="button-brand" kxType="ripple" disabled>
        Kinetics Brand Button
      </Button>
    )
  },
  {
    id: 'outline-brand-disabled',
    label: 'Outline Brand - Disabled',
    element: (
      <Button isOutlineBrand disabled>
        Outline Brand Button
      </Button>
    )
  },
  {
    id: 'outline-brand-kinetics-disabled',
    label: 'Outline Brand (Kinetics) - Disabled',
    element: (
      <Button isOutlineBrand kxScope="button-outline" kxType="ripple" disabled>
        Kinetics Outline Brand Button
      </Button>
    )
  },
  {
    id: 'inverse-disabled',
    label: 'Inverse - Disabled',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button isInverse disabled>
          Inverse Button
        </Button>
      </div>
    )
  },
  {
    id: 'destructive-disabled',
    label: 'Destructive - Disabled',
    element: (
      <Button isDestructive disabled>
        Destructive Button
      </Button>
    )
  },
  {
    id: 'text-destructive-disabled',
    label: 'Text Destructive - Disabled',
    element: (
      <Button isTextDestructive disabled>
        Text Destructive Button
      </Button>
    )
  },
  {
    id: 'success-disabled',
    label: 'Success - Disabled',
    element: (
      <Button isSuccess disabled>
        Success Button
      </Button>
    )
  }
];
