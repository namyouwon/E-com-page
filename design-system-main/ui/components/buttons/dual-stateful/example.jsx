// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../../../shared/svg-icon';
import classNames from 'classnames';
import { Button } from '../base/example';

const DefaultText = props => {
  return (
    <span className="slds-text-not-pressed">
      <SvgIcon
        className="slds-button__icon slds-button__icon_small slds-button__icon_left"
        sprite="utility"
        symbol="add"
      />
      {props.children}
    </span>
  );
};

DefaultText.propTypes = {
  children: PropTypes.string.isRequired
};

const PressedText = props => {
  return (
    <span className="slds-text-pressed">
      <SvgIcon
        className="slds-button__icon slds-button__icon_small slds-button__icon_left"
        sprite="utility"
        symbol="check"
      />
      {props.children}
    </span>
  );
};

PressedText.propTypes = {
  children: PropTypes.string.isRequired
};

const DualStatefulButton = props => {
  const { isDisabled, isPressed } = props;
  const [buttonSelected, setButtonSelected] = useState(isPressed);

  return (
    <Button
      isNeutral
      className={classNames('slds-button_dual-stateful', {
        'slds-is-pressed': buttonSelected
      })}
      disabled={isDisabled}
      aria-live="assertive"
      aria-pressed={Boolean(buttonSelected).toString()}
      onClick={() => setButtonSelected(!buttonSelected)}
    >
      <DefaultText>Follow</DefaultText>
      <PressedText>Following</PressedText>
    </Button>
  );
};

DualStatefulButton.propTypes = {
  isDisabled: PropTypes.bool,
  isPressed: PropTypes.bool
};

export { DualStatefulButton, DefaultText, PressedText };

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <DualStatefulButton>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    )
  }
];

export const states = [
  {
    id: 'disabled',
    label: 'disabled',
    element: (
      <DualStatefulButton isDisabled>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    )
  }
];

export const examples = [
  {
    id: 'pressed',
    label: 'pressed',
    element: (
      <DualStatefulButton isPressed>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    )
  }
];
