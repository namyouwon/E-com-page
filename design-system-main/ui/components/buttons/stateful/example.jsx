// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../../../shared/svg-icon';
import classNames from 'classnames';
import { Button } from '../base/example';

export let StatefulButton = props => {
  const {
    className,
    disabled,
    isNotSelected,
    isSelected,
    isSelectedClicked
  } = props;
  const [buttonSelected, setButtonSelected] = useState(
    (typeof isNotSelected !== 'undefined' &&
      Boolean(isNotSelected) === false) ||
      isSelected ||
      isSelectedClicked
  );
  const [buttonFocused, setButtonFocused] = useState(isSelectedClicked);

  return (
    <Button
      className={classNames('slds-button_stateful', className, {
        'slds-not-selected': !buttonSelected,
        'slds-is-selected': buttonSelected && !buttonFocused,
        'slds-is-selected-clicked': buttonSelected && buttonFocused
      })}
      disabled={disabled}
      aria-live="assertive"
      aria-pressed={Boolean(buttonSelected).toString()}
      isNeutral
      onBlur={() => setButtonFocused(false)}
      onFocus={() => setButtonFocused(true)}
      onClick={() => setButtonSelected(!buttonSelected)}
    >
      <span className="slds-text-not-selected">
        <SvgIcon
          className="slds-button__icon slds-button__icon_small slds-button__icon_left"
          sprite="utility"
          symbol="add"
        />
        Follow
      </span>
      <span className="slds-text-selected">
        <SvgIcon
          className="slds-button__icon slds-button__icon_small slds-button__icon_left"
          sprite="utility"
          symbol="check"
        />
        Following
      </span>
      <span className="slds-text-selected-focus">
        <SvgIcon
          className="slds-button__icon slds-button__icon_small slds-button__icon_left"
          sprite="utility"
          symbol="close"
        />
        Unfollow
      </span>
    </Button>
  );
};

StatefulButton.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  isNotSelected: PropTypes.bool,
  isSelected: PropTypes.bool,
  isSelectedClicked: PropTypes.bool
};

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default <StatefulButton isNeutral isNotSelected />;

export const states = [
  {
    id: 'disabled',
    label: 'disabled',
    element: <StatefulButton isNotSelected disabled />
  },
  {
    id: 'selected',
    label: 'Selected',
    element: <StatefulButton isNeutral isSelected />
  },
  {
    id: 'selected-clicked',
    label: 'Selected and Clicked',
    element: <StatefulButton isNeutral isSelectedClicked />
  }
];
