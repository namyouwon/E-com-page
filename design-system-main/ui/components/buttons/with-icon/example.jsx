// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';
import { Button } from '../base/example';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Button isNeutral>
        <SvgIcon
          className="slds-button__icon slds-button__icon_left"
          sprite="utility"
          symbol="download"
        />
        Button Neutral
      </Button>
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Disabled',
    element: (
      <Button isNeutral disabled>
        <SvgIcon
          className="slds-button__icon slds-button__icon_left"
          sprite="utility"
          symbol="download"
        />
        Button Neutral
      </Button>
    )
  }
];

export let examples = [
  {
    id: 'with-icon-right',
    label: 'With right icon',
    element: (
      <Button isNeutral>
        Button Neutral
        <SvgIcon
          className="slds-button__icon slds-button__icon_right"
          sprite="utility"
          symbol="download"
        />
      </Button>
    )
  }
];
