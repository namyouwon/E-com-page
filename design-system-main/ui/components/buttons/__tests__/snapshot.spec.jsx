/* eslint-env jest */
import React from 'react';
import { Button } from '../base/example';
import { StatefulButton } from '../stateful/example';
import {
  DualStatefulButton,
  DefaultText,
  PressedText
} from '../dual-stateful/example';
import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

describe('Button ', () => {
  it('renders a base button', () => matchesMarkup(<Button>Button</Button>));

  it('renders a disabled base button', () =>
    matchesMarkup(<Button disabled>Button</Button>));

  it('renders a neutral button', () =>
    matchesMarkup(<Button isNeutral>Button</Button>));

  it('renders a brand button', () =>
    matchesMarkup(<Button isBrand>Button</Button>));

  it('renders an outline brand button', () =>
    matchesMarkup(<Button isOutlineBrand>Button</Button>));

  it('renders an inverse button', () =>
    matchesMarkup(<Button isInverse>Button</Button>));

  it('renders a desctructive button', () =>
    matchesMarkup(<Button isDestructive>Button</Button>));

  it('renders a text destructive button', () =>
    matchesMarkup(<Button isTextDestructive>Button</Button>));

  it('renders a success button', () =>
    matchesMarkup(<Button isSuccess>Button</Button>));

  it('renders a full width, reset button', () =>
    matchesMarkup(<Button isFullWidth>Button</Button>));

  it('renders a stretched button', () =>
    matchesMarkup(
      <Button isSuccess isStretch>
        Button
      </Button>
    ));

  it('renders a link that looks like a button', () =>
    matchesMarkup(
      <Button use="a" isSuccess>
        Link as a Button
      </Button>
    ));
});

describe('Stateful Button ', () => {
  it('renders a not-selected stateful button', () =>
    matchesMarkup(<StatefulButton isNotSelected>Button</StatefulButton>));

  it('renders a selected and focused stateful button', () =>
    matchesMarkup(<StatefulButton isSelectedClicked>Button</StatefulButton>));

  it('renders a selected stateful button', () =>
    matchesMarkup(<StatefulButton isSelected>Button</StatefulButton>));
});

describe('Dual Stateful Button', () => {
  it('renders a not-pressed dual stateful button', () =>
    matchesMarkup(
      <DualStatefulButton>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    ));

  it('renders a pressed dual stateful button', () =>
    matchesMarkup(
      <DualStatefulButton isPressed>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    ));

  it('renders a disabled dual stateful button', () =>
    matchesMarkup(
      <DualStatefulButton isDisabled>
        <DefaultText>Follow</DefaultText>
        <PressedText>Following</PressedText>
      </DualStatefulButton>
    ));
});
