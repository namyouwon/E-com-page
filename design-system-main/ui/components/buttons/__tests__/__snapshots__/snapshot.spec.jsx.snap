// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  renders a base button 1`] = `
<button
  className="slds-button"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a brand button 1`] = `
<button
  className="slds-button slds-button_brand"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a desctructive button 1`] = `
<button
  className="slds-button slds-button_destructive"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a disabled base button 1`] = `
<button
  className="slds-button"
  disabled={true}
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a full width, reset button 1`] = `
<button
  className="slds-button slds-button_full-width"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a link that looks like a button 1`] = `
<a
  className="slds-button slds-button_success"
  href="#"
  onClick={[Function]}
>
  Link as a Button
</a>
`;

exports[`Button  renders a neutral button 1`] = `
<button
  className="slds-button slds-button_neutral"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a stretched button 1`] = `
<button
  className="slds-button slds-button_success slds-button_stretch"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a success button 1`] = `
<button
  className="slds-button slds-button_success"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders a text destructive button 1`] = `
<button
  className="slds-button slds-button_text-destructive"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders an inverse button 1`] = `
<button
  className="slds-button slds-button_inverse"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Button  renders an outline brand button 1`] = `
<button
  className="slds-button slds-button_outline-brand"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`Dual Stateful Button renders a disabled dual stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="false"
  className="slds-button slds-button_neutral slds-button_dual-stateful"
  disabled={true}
  onClick={[Function]}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Dual Stateful Button renders a not-pressed dual stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="false"
  className="slds-button slds-button_neutral slds-button_dual-stateful"
  onClick={[Function]}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Dual Stateful Button renders a pressed dual stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="true"
  className="slds-button slds-button_neutral slds-button_dual-stateful slds-is-pressed"
  onClick={[Function]}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Stateful Button  renders a not-selected stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="false"
  className="slds-button slds-button_neutral slds-button_stateful slds-not-selected"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;

exports[`Stateful Button  renders a selected and focused stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="true"
  className="slds-button slds-button_neutral slds-button_stateful slds-is-selected-clicked"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;

exports[`Stateful Button  renders a selected stateful button 1`] = `
<button
  aria-live="assertive"
  aria-pressed="true"
  className="slds-button slds-button_neutral slds-button_stateful slds-is-selected"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;
